app/src/app/components/add-relationship-button/add-relationship-button.component.ts

```diff
@@ -24,33 +25,41 @@ export class AddRelationshipButtonComponent implements OnInit {
         let relationship = new Relationship();
         relationship.relationship_type = this.config.relationship_type;
         let initializer = null;
-        if (this.config.source_object) initializer = relationship.set_source_object(this.config.source_object);
+        if (this.config.source_object) initializer = relationship.set_source_object(this.config.source_object, this.restApiService);
         else if (this.config.source_ref) initializer = relationship.set_source_ref(this.config.source_ref, this.restApiService);
-        else if (this.config.target_object) initializer = relationship.set_target_object(this.config.target_object);
+        else if (this.config.target_object) initializer = relationship.set_target_object(this.config.target_object, this.restApiService);
         else if (this.config.target_ref) initializer = relationship.set_target_ref(this.config.target_ref, this.restApiService);
         else initializer = EMPTY;
         this.loading = true;
         var zip_subscription = initializer.subscribe({
             next: () => {
                 this.loading = false;
-                let prompt = this.dialog.open(StixDialogComponent, {
-                    data: {
-                        object: relationship,
-                        editable: true,
-                        mode: "edit",
-                        sidebarControl: "events"
-                    },
-                    maxHeight: "75vh"
-                })
-                let subscription = prompt.afterClosed().subscribe({
-                    next: result => { if (prompt.componentInstance.dirty) this.created.emit(); }, //re-fetch values since an edit occurred
-                    complete: () => { subscription.unsubscribe(); }
-                });
+                let config: StixViewConfig = {
+                    object: relationship,
+                    editable: true,
+                    mode: "edit",
+                    sidebarControl: "events"
+                }
+
+                if (this.config.dialog && this.config.dialog.componentInstance) {
+                    // replace current dialog content
+                    this.config.dialog.componentInstance._config = config;
+                    this.config.dialog.componentInstance.startEditing();
```

Comment:
This works but is a bit hard to follow as a reader of the code. Might want to expand on the documentation (here and the other places this feature is found): 
- When is the param relevant? (when you want to add a relationship while working inside of a dialog)
- What does it do (you already explain this, but maybe could expand on it a bit more)